crates/sui-types/src/inner_temporary_store.rs

```diff
@@ -34,6 +34,8 @@ pub struct InnerTemporaryStore {
     pub binary_config: BinaryConfig,
     pub runtime_packages_loaded_from_db: BTreeMap<ObjectID, PackageObject>,
     pub lamport_version: SequenceNumber,
+    // Mapping of the mutated config objects to their new sequence numbers at the time of mutation.
+    pub mutated_config_objects: BTreeMap<ObjectID, SequenceNumber>,
```

Comment:
Do you have to pass it through from the temp store? Could you compute it in the transaction output instead?

Reply:
+1

Reply:
I initially started out with it in transaction output computation however it felt a bit odd to me to do it there (with storage reading and the like). It felt a bit better here and I like having the computation for both the loaded config objects and mutated config objects in one place (loaded config object stuff will need to stay in `into_effects` due to them needing to be passed to effects generation). 

That being said I don't have a super strong opinion on it, so can move it y'all would prefer.